### PR TITLE
Added cli cert command, changed keylisting to be a map

### PR DESCRIPTION
**Pull request by diogomonica**
- Added a new cli cert command
- Changed the keylisting to be a map instead of list
- Fixed removal of normal keys, and force root keys deletion to need -r

Signed-off-by: Diogo Monica <diogo@docker.com>

### CODE REVIEW

File: cmd/notary/cert.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+			fatalf("unable to retrieve certificate with cert ID: %s", certID)
+		}
+		certsToRemove = append(certsToRemove, cert)
+	} else {
+		// We got the -g flag, it's a GUN
+		certs, err := keyStoreManager.TrustedCertificateStore().GetCertificatesByCN(certRemoveGUN)
+		if err != nil {
+			fatalf("%v", err)
+		}
+		certsToRemove = append(certsToRemove, certs...)
```
Comment (aaronlehmann) [inline cmd/notary/cert.go]: Looks like we ignore `args[0]` in this case. We should check if both `-g` and the positional argument are specified, and report an error.

File: cmd/notary/cert.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+	// List all the keys about to be removed
+	fmt.Println("Are you sure you want to remove the following certificates?")
+	for _, cert := range certsToRemove {
+		// This error can't occur because we're getting certs off of an
+		// x509 store that indexes by ID.
+		certID, _ := trustmanager.FingerprintCert(cert)
+		fmt.Printf("%s - %s\n", cert.Subject.CommonName, certID)
+	}
+	fmt.Println("(yes/no)")
```
Comment (aaronlehmann) [inline cmd/notary/cert.go]: I'd change the first Println to say "The following certificates will be removed:" and the second one to say "Are you sure you want to remove these certificates? (yes/no)". But feel free to ignore... this is a matter of opinion.

File: cmd/notary/cert.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+func printCert(cert *x509.Certificate) {
+	timeDifference := cert.NotAfter.Sub(time.Now())
+	certID, err := trustmanager.FingerprintCert(cert)
+	if err != nil {
+		fatalf("could not fingerprint certificate: %v", err)
+	}
+
+	fmt.Printf("%s %s (expires in: %v days)\n", cert.Subject.CommonName, certID, math.Floor(timeDifference.Hours()/24))
```
Comment (aaronlehmann) [inline cmd/notary/cert.go]: What if it's expired? Would we print a negative expiration time? Or is there something in the cert loading path that filters out expired certificates? Normally I'd worry about a race here, when we load a non-expired certificate and then it expires before we do this calculation, but we're using units of days here so I don't think we need to worry.

**Comment by aaronlehmann** [comment]
Noted a few things, but looks good to me in general. I'd suggest checking for conflicting arguments in certRemove, but the other suggestions are optional.

File: cmd/notary/cert.go (hunk 3 above)
Comment (diogomonica) [inline cmd/notary/cert.go]: We never have expired certs loaded into keystore. We always create FilteredX509FileStores.

File: cmd/notary/cert.go (hunk 1 above)
Comment (diogomonica) [inline cmd/notary/cert.go]: Added

**Pull request merged**